### PR TITLE
Bugfix/168 account creation

### LINKED ISSUE #168: Account creation bugs

### PR DESCRIPTION
**Pull request by pierreTklein**
# Description

Fixed various bugs that arose during the account creation process:
- returned hacker object contains `_id`, not `id`.
- hacker error message was 500, but status was 400.
- wrapped `createRoleBindings` with async middleware
- changed `ethnicity` attribute to array to accommodate for multiple ethnicities.
- generalized integerValidator error messages
- removed `required: true` field in schema
- Fixed invalid test

Fixes #168

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?

Tested via frontend.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: middlewares/auth.middleware.js [hunk 1]
```diff
@@ -391,22 +391,10 @@ async function addCreationRoleBindings(req, res, next) {
  * @param {string} roleName name of the role to be added to account
  */
 function createRoleBindings(roleName = undefined) {
-    return async (req, res, next) => {
+    return Middleware.Util.asyncMiddleware(async (req, res, next) => {
         await Services.RoleBinding.createRoleBindingByRoleName(req.user.id, roleName);
```
Comment (pashakondratyev) [inline middlewares/auth.middleware.js]: add await Services.RoleBinding.createRoleBindingByRoleName(req.body.account.id, "account");

File: middlewares/auth.middleware.js [hunk 2]
```diff
@@ -438,7 +414,6 @@ module.exports = {
     validateConfirmationTokenWithoutAccount: Middleware.Util.asyncMiddleware(validateConfirmationTokenWithoutAccount),
     createRoleBindings: createRoleBindings,
     addCreationRoleBindings: Middleware.Util.asyncMiddleware(addCreationRoleBindings),
-    addSponsorRoleBindings: Middleware.Util.asyncMiddleware(addSponsorRoleBindings),
     resendConfirmAccountEmail: Middleware.Util.asyncMiddleware(resendConfirmAccountEmail),
     retrieveRoleBindings: Middleware.Util.asyncMiddleware(retrieveRoleBindings),
     retrieveRoles: Middleware.Util.asyncMiddleware(retrieveRoles)
```
[the remark this answers is not in the document]
Comment (pierreTklein) [reply middlewares/auth.middleware.js]: retrieveRoles?

[the remark this answers is not in the document]
Comment (pashakondratyev) [reply middlewares/auth.middleware.js]: Yeah, sorry wasn't clear but you deleted the retrieveRoles function

[the remark this answers is not in the document]
Comment (pierreTklein) [reply middlewares/auth.middleware.js]: oh, that shouldn't be the case :/

[the remark this answers is not in the document]
Comment (pierreTklein) [reply middlewares/auth.middleware.js]: Hmm I don't think I deleted them

**Pull request merged**